prowler/providers/aws/lib/service/service.py

```diff
@@ -21,6 +21,17 @@ class AWSService:
     - Also handles if the AWS Service is Global
     """
 
+    failed_checks = set()
+
+    @classmethod
+    def set_failed_check(cls, check_id=None, resource_id=None, region=None):
+        if check_id is not None and resource_id is not None and region is not None:
+            cls.failed_checks.add((check_id.split(".")[-1], resource_id, region))
```

Comment:
> Why do you need the `split(".")`?

In order to split the class name from the path. 
Even though both Getter and Setter method pass the class name (instead of the full qualified name which contains the path), the Get method passes the qualified name while the Set method passes the class name.
This is how the value is set:
```
ec2_client.set_failed_check(
    self.__class__.__name__,
    security_group.id,
    security_group.region,
)
```
While this is how the value is retrieved:

```
ec2_client.is_failed_check(
    ec2_securitygroup_allow_ingress_from_internet_to_all_ports.__name__,
    security_group.id,
    security_group.region,
)
```
As you can see, both are using `__name__` and not `__qualname__`. However, The Get method seems to pass the full qualified class name which contains the path. 
To resolve it, it's cut by the `split(".")` method, so we are sure to have the class name without path in all cases. If you have a better solution, let me know please.

Reply:
> We need to add tests to this functions in the `TestAWSService` class.

Ok, no problem.